tensorflow/stream_executor/stream.h

```diff
@@ -628,43 +628,49 @@ class Stream {
                           const dnn::BatchDescriptor &input_dimensions,
                           const DeviceMemory<double> &input_data,
                           const dnn::BatchDescriptor &output_dimensions,
-                          DeviceMemory<double> *output_data);
+                          DeviceMemory<double> *output_data,
+                          ScratchAllocator *workspace_allocator = nullptr);
```

Comment:
Shouldn't these changes not be present in this commit?

Reply:
This is made intentionally to make sure subsequent PRs such as #20713 be built properly.

Reply:
Part of the purpose of splitting out the patches is to make subsequent patches easier to read.  If we combine all of the dependent patches in here, then this one is hard to read, defeating most of the purpose.

Can you include the other, split-out commits, in this PR?  Then the top commit in this PR won't contain the commits you split out, and we can just review the top commit.

Alternatively I guess you could just send one mega PR with multiple commits.  I've never done a big review in Github, so I'm not sure which would be easier; I guess it's up to you.  The important thing is that each atomic change is split out and individually documented, because that makes it much easier to review.

Reply:
Thanks for the feedback. Let me see how to refactor this PR.